src/services/iofog-service.js

```diff
@@ -0,0 +1,228 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2018 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+const TransactionDecorator = require('../decorators/transaction-decorator')
+const AppHelper = require('../helpers/app-helper')
+const FogManager = require('../sequelize/managers/iofog-manager')
+const FogProvisionKeyManager = require('../sequelize/managers/iofog-provision-key-manager')
+const FogVersionCommandManager = require('../sequelize/managers/iofog-version-command-manager')
+const ChangeTrackingManager = require('../sequelize/managers/change-tracking-manager')
+const Errors = require('../helpers/errors')
+const ObjBuilder = require('../helpers/object-builder')
+
+async function _createFog(fogData, user, transaction) {
+  AppHelper.validateFields(fogData, ['name', 'fogType'])
+  _validateLatLon(fogData.latitude, fogData.longitude)
+
+  const ob = new ObjBuilder()
+  const createFogData = ob
+    .pushRequiredFieldWithCondition('uuid', fogData.uuid, fogData.uuid, AppHelper.generateRandomString(32))
```

Comment:
in refactoring now. will be simplified to:
const  createFogData = {
   filed: fogData.field
   otherField: fogData.otherField
   ...
}
AppHelper.deleteUndefinedFields(createFogData)